components/renku_data_services/users/blueprints.py

```diff
@@ -84,12 +80,7 @@ async def _get_one(request: Request, user: base_models.APIUser, user_id: str) ->
             if not user_info:
                 raise errors.MissingResourceError(message=f"The user with ID {user_id} cannot be found.")
             return json(
-                {
-                    "id": user_info.id,
-                    "first_name": user_info.first_name,
-                    "last_name": user_info.last_name,
-                    "email": user_info.email,
-                }
+                apispec.UserWithId.model_validate(user_info.to_renku_user()).model_dump(exclude_none=True, mode="json")
             )
```

Comment:
Not respecting an API contract is **worse** than responding with an error.

If the response doesn't conform to the API spec, the UI will hard crash; while if the response is a 422, the user can still use the app.

Reply:
Response validation happens in:
* components/renku_data_services/namespace/blueprints.py
* components/renku_data_services/storage/blueprints.py
* components/renku_data_services/users/blueprints.py (UserSecretsBP)

All of which *I* did not write. I do not see why using the same approach as other modules in the code base should be avoided here.

Reply:
Seems to me the proper solution would be to keep validation but make it so the status code is 500 if it fails?

Reply:
It should already happen in the `db.py` code anyway (a 500 error is raised if the namespace doesn't exist). I could make an issue to standardize response validation error codes.

Reply:
it'd need a bit more to work automatically (or lot's of try/except :frowning_face: ) so we have 422 if the request doesn't validate but 500 for invalid responses.

Reply:
We can have a helper like so:

```python
"""Base response validation used by services."""

from collections.abc import Callable
from typing import Any

from pydantic import BaseModel
from pydantic import ValidationError as PydanticValidationError
from sanic import json
from sanic.response import JSONResponse

from renku_data_services import errors


def validated_json(
    model: type[BaseModel],
    data: Any,
    exclude_none: bool = True,
    status: int = 200,
    headers: dict[str, str] | None = None,
    content_type: str = "application/json",
    dumps: Callable[..., str] | None = None,
    **kwargs: Any,
) -> JSONResponse:
    """Creates a JSON response with data validation."""
    try:
        body = model.model_validate(data).model_dump(exclude_none=exclude_none, mode="json")
    except PydanticValidationError as err:
        raise errors.ProgrammingError from err
    return json(body, status=status, headers=headers, content_type=content_type, dumps=dumps, **kwargs)
```

and then we replace `json(...)` with `validated_json(Model, data, ...)`

Reply:
> Response validation happens in:
>    components/renku_data_services/namespace/blueprints.py
>    components/renku_data_services/storage/blueprints.py
>    components/renku_data_services/users/blueprints.py (UserSecretsBP)
> All of which I did not write. I do not see why using the same approach as other modules in the code base should be avoided here.

@leafty I know, I wrote some of them and I realized later that the approach is problematic and confusing for users. But I dont really want to use this more so that is why I asked you to modify it in this PR. 

If you add the `validated_json` function in this PR I will onboard the existing instances to this function later. 

I opened an issue for this here: https://github.com/SwissDataScienceCenter/renku-data-services/issues/266

Reply:
Or you can also leave everything as is and we will fix the response codes later during the cooldown.

Reply:
Done in a7c0fba7b19249620a1d1b79756dbcdf11af4827.

I checked locally that creating an invalid response (e.g. removing a field or adding one) will create a 500 response and fail corresponding tests in `test/bases/renku_data_services/data_api/`.

This PR is not going to be merged to `main`, so if desired I can also cherry-pick the `validated_json()` method in a new PR.

I think that this should probably be the standard way to construct API responses so that we ensure the API contract in all endpoints.

Reply:
This will eventually end up in main that is ok.